### ISSUE TITLE
Inline errors do not clear after selection for demographics questions in patient registration form

### ISSUE THREAD

**Issue by emyl3** [CDCgov/prime-simplereport#8548]
## Area or Component

Patient registration form - Demographics section

## Quick Summary

Inline error messages for the "Demographics" section of the patient registration form do not clear after an option is selected

### How to Replicate

- Go to the `/register/<pxp code>` page and navigate to the patient registration form
- Fill in the first name or any other field excluding the "Demographics" inputs
- Click "Submit"
- Notice that the inline errors appear for the "Demographics" section for the following fields: Race, Are you Hispanic or Latino?, Sex
- Make the corrections and click "Submit" again
- The inline errors do not clear

## Accessibility Violation

This is misleading to the end user trying to complete the form

## Action Requested

- Clear the inline errors when a selection is made for the inputs under the "Demographics" section
- Update the inline error for "Sex" to use the word "sex"

## Screenshots (optional)

<img width="663" alt="Image" src="https://github.com/user-attachments/assets/c5f48442-5c75-45af-a3fb-2e76347403e0" />
